core/block/editor/basic/basic.go

```diff
@@ -278,6 +287,48 @@ func (bs *basic) Move(srcState, destState *state.State, targetBlockId string, po
 	return srcState.InsertTo(targetBlockId, position, blockIds...)
 }
 
+func checkTableBlocksMove(st *state.State, target string, pos model.BlockPosition, blockIds []string) (string, model.BlockPosition, error) {
+	for _, id := range blockIds {
+		root := table.GetTableRootBlock(st, id)
+		if root != nil && root.Model().Id != id {
+			t, err := table.NewTable(st, root.Model().Id)
+			if err != nil {
+				return "", 0, ErrCannotMoveTableBlocks
+			}
+
+			// we allow moving rows between each other
+			rows := t.RowIDs()
+			if slices.Contains(rows, id) && slices.Contains(rows, target) {
+				if pos == model.Block_Bottom || pos == model.Block_Top {
+					return target, pos, nil
```

Comment:
Probably you don't want to return here(and other places)  in case of success? Because you are iterating the slice of BlockIds

Reply:
Fixed